pkg/engine/utils.go

```diff
@@ -49,6 +49,15 @@ func checkNameSpace(namespaces []string, resourceNameSpace string) bool {
 	return false
 }
 
+func checkAnnotations(annotations map[string]string, resourceAnnotations map[string]string) bool {
+	for k, v := range annotations {
+		if resourceAnnotations[k] != v {
```

Comment:
It would be safer to add a check on `resourceAnnotations` as it could be an empty map.
So before iterating through annotations, add 
````
if resourceAnnotations == nil {
  //  return
}
````

Reply:
updated accordingly